lib/active_model/serializer/adapter.rb

```diff
@@ -103,39 +107,8 @@ def fragment_cache(*args)
 
       private
 
-      def cache_check(serializer)
-        @cached_serializer = serializer
-        @klass             = @cached_serializer.class
-        if is_cached?
-          @klass._cache.fetch(cache_key, @klass._cache_options) do
-            yield
-          end
-        elsif is_fragment_cached?
-          FragmentCache.new(self, @cached_serializer, @options).fetch
-        else
-          yield
-        end
-      end
-
-      def is_cached?
-        @klass._cache && !@klass._cache_only && !@klass._cache_except
-      end
-
-      def is_fragment_cached?
-        @klass._cache_only && !@klass._cache_except || !@klass._cache_only && @klass._cache_except
-      end
-
-      def cache_key
-        parts = []
-        parts << object_cache_key
-        parts << @klass._cache_digest unless @klass._cache_options && @klass._cache_options[:skip_digest]
-        parts.join('/')
-      end
-
-      def object_cache_key
-        object_time_safe = @cached_serializer.object.updated_at
-        object_time_safe = object_time_safe.strftime('%Y%m%d%H%M%S%9N') if object_time_safe.respond_to?(:strftime)
-        (@klass._cache_key) ? "#{@klass._cache_key}/#{@cached_serializer.object.id}-#{object_time_safe}" : @cached_serializer.object.cache_key
+      ActiveModelSerializers.silence_warnings do
+        attr_reader :options
```

Comment:
I'm thinking we should rename this to `instance_options` so that we can replace the `@options` with `instance_options` and don't need to change any method arguments from `options` to e.g. `opts`.. but it seems.. but not in this PR, mostly because I might not have time to do it right